mage_ai/streaming/sinks/influxdb.py

```diff
@@ -0,0 +1,95 @@
+import time
+from collections.abc import Iterable
+from dataclasses import dataclass
+from typing import Dict, List
+
+from influxdb_client import InfluxDBClient, WriteOptions, WritePrecision
+
+from mage_ai.shared.config import BaseConfig
+from mage_ai.streaming.constants import DEFAULT_BATCH_SIZE, DEFAULT_TIMEOUT_MS
+from mage_ai.streaming.sinks.base import BaseSink
+
+
+def get_timestamp_ms():
+    return int(time.time() * 1e3)
+
+
+@dataclass
+class InfluxDbConfig(BaseConfig):
+    url: str
+    token: str
+    org: str
+    bucket: str = "data"
+    measurement: str = "default"
+    batch_size: int = DEFAULT_BATCH_SIZE
+    timeout_ms: int = DEFAULT_TIMEOUT_MS
+
+
+class InfluxDbSink(BaseSink):
+    config_class = InfluxDbConfig
+
+    def init_client(self):
+        self._print("Start initializing writer.")
+        # Initialize influxdb client and write_api
+        self.client = InfluxDBClient(
+            url=self.config.url,
+            token=self.config.token,
+            org=self.config.org,
+        )
+        if self.config.batch_size > 0:
+            batch_size = self.config.batch_size
+        else:
+            batch_size = DEFAULT_BATCH_SIZE
+        if self.config.timeout_ms > 0:
+            timeout_ms = self.config.timeout_ms
+        else:
+            timeout_ms = DEFAULT_TIMEOUT_MS
+
+        options = WriteOptions(
+            batch_size=batch_size,
+            flush_interval=timeout_ms,
+            jitter_interval=0,
+            retry_interval=5_000,
+            max_retries=5,
+            max_retry_delay=125_000,
+            exponential_base=2,
+        )
+        self.write_api = self.client.write_api(write_options=options)
+        self._print("Finish initializing writer.")
+
+    def write(self, data: Dict):
```

Comment:
To include the metadata in the data, I would update the data format to be
```
{
    "record": {...},
    "metadata": {...}
}
``

Reply:
I am not used to github, so please excuse my confusion. 
If you are referring to line 57: I think the best place for write_options is the config yaml.
If you are referring to line 60:

Yes that's a way to structure the data. I'am not sure why you call it record and metadata, as record suggests it includes metadata itself. I would suggest data and metadata instead, i.e.

```python
message = {"data": {...}, "metadata": {...}}
```

But that's personal preference. A typical message to influxdb would then look like this:

```python
message = {
  "data": {"bees": 23}
  "metadata": {
    "measurement": "census",
    "time": 1693472675783,
    "tags": {"location": "klamath", "scientist": "anderson"}
  }
}
```

So basically influxdb "fields" can be found in the "data" section, everything else is in "metadata" and has defaults if not given explicitly. I am currently preparing a PR for including metadata in kafka sink and source. A typical message from kafka source could look like this:

```python
message = {
  "data": {
    "bees": {
      "location": "klamath", 
      "scientist": "anderson", 
      "count": 23
    },
  "metadata": {
    "topic": "census",
    "partition": 0,
    "offset": 6,
    "time": 1693472675783
  }
```

Ultimately all sinks and sources should share some basic interpretation of the messages, e.g. time stamps from kafka should be used when writing them to influxdb without the need to restructure the data. Specialties like the distinction between tags and fields in InfluxDB can't or shouldn't be handled automatically (IMHO).

I will update my code to match the new schema.

Reply:
Yes, that makes sense. We can use this format.